UnitTests/GitUITests/SpellChecker/WordAtCursorExtractorTests.cs

```diff
@@ -0,0 +1,30 @@
+﻿using GitUI.SpellChecker;
+using NUnit.Framework;
+
+namespace GitUITests.SpellChecker
+{
+    [TestFixture]
+    public class WordAtCursorExtractorTests
+    {
+        private WordAtCursorExtractor _wordAtCursorExtractor;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _wordAtCursorExtractor = new WordAtCursorExtractor();
+        }
+
+        [TestCase(".git", ExpectedResult = ".git")]
+        [TestCase("Add .g", ExpectedResult = ".g")]
+        [TestCase("Add.g", ExpectedResult = "g")]
+        [TestCase("Add,git", ExpectedResult = "git")]
+        [TestCase("Add (.git", ExpectedResult = ".git")]
+        [TestCase("Introduce Tes", ExpectedResult = "Tes")]
+        [TestCase("Add).git", ExpectedResult = ".git")]
+        [TestCase("[Add].git", ExpectedResult = ".git")]
+        public string ExtractsMeaningfulWord(string text)
+        {
+            return _wordAtCursorExtractor.Extract(text, text.Length - 1);
```

Comment:
What is the use case for these:
```
[TestCase("Add).git", ExpectedResult = ".git")]
[TestCase("[Add].git", ExpectedResult = ".git")]
```
Should not they be:
```
[TestCase("Add).git", ExpectedResult = "git")]
[TestCase("[Add].git", ExpectedResult = "git")]
```
the same as it is for:
```
[TestCase("Add.g", ExpectedResult = "g")]
```
?

Reply:
A case would be like `[#5555].gitignore was added`. Of course it is a bit synthetical case, but in general my point was that, when I need to choose between a word and a word starting with a dot, I choose the last. I hope that a user will use space or other separator (except dot) if he wants to type a word which does not start with a dot. Otherwise I consider he wants to type a word, which starts with a dot.

Reply:
It looks like there is a space missing after the ']'
I think that we could treat ')' and ']' as non separator characters because they are often used in code like `func().otherFunc()` or `array[0].func()`

Reply:
I have changed the behavior for the cases when ']' and ')' are before the dot.